modules/guardrails-autoconfiguration.adoc

```diff
@@ -0,0 +1,43 @@
+[id='auto-configuration-guardrails-orchestrator-service_{context}']
+
+= Auto-configuring Guardrails
+
+[role='_abstract']
+Auto-configuration simplifies the guardrails setup process by automatically handling TLS configuration and authentication, ensuring seamless and secure communication between components. Use the `autoConfig` specification in the `GuardrailsOrchestrator` custom resource (CR) to integrate with detector services in your namespace, creating seamless and secure communication between components.
+For example, if any of the detector or generation services use HTTPS, their credentials are automatically discovered, mounted, and used. Additionally, the orchestrator is automatically configured to forward all necessary authentication token headers.
+
+.Prerequisites
+* Each detector server you intend to use has an OpenShift label applied, such as _<label_name>_: true. Choose a descriptive name for the label as it is required for configuration.
+* An inference service to which you intend to apply guardrails.
+
+.Procedure
+* Create a  `GuardrailsOrchestrator` CR with the `autoConfig` configuration:
++
+[source,yaml]
+----
+apiVersion: trustyai.opendatahub.io/v1alpha1
+kind: GuardrailsOrchestrator
+metadata:
+  name: guardrails-orchestrator
+  annotations:
+    security.opendatahub.io/enable-auth: 'true'
+spec:
+  autoConfig:
+      inferenceServiceToGuardrail: $INFERENCE_SERVICE_NAME
+      detectorServiceLabelToMatch: $LABEL_NAME
+  enableBuiltInDetectors: true
+  enableGuardrailsGateway: true 
+  replicas: 1
+----
++
+where:
+
+inferenceServiceToGuardrail:: specifies the name of the vLLM inference service to be guardrailed.
+detectorServiceLabelToMatch:: specifies the label that we chose to apply to each of our detector servers.
+
+.Verification
+After you create the GuardrailsOrchestrator CR, the system creates the following resources:
+
+* An automatically generated orchestrator configuration config map. This configuration automatically updates to reflect the state of detectors in your namespace.
+
+* An automatically generated default gateway config map, if enableGuardrailsGateway is set to true. You can use this config map as a starting template to build a custom gateway configuration. To restore the default gateway configuration, delete the config map, and a new one is created automatically.
```

Comment:
@kpunwatk Is there any code a user would run in the command line to view these resources?